clients/cobol-lsp-vscode-extension/src/services/copybook/InfoStorage.ts

```diff
@@ -0,0 +1,18 @@
+import { SettingsService } from "../Settings";
+
+export class InfoStorage {
+    private static readonly map = new Map<string, string>();
+
+    public static set(cobolFileName: string, copybookName: string, dialectType: string) {
+        InfoStorage.map.set(`${cobolFileName}#${copybookName}`, dialectType);
```

Comment:
Unfortunately, it will not work in the case when we have the same-named copybooks from different dialects.
Yes, it is not working even right now due to caching in the server. But it will be better to implement this support right now rather than fix this in the server and here in the client.

Reply:
I am working on it, it is simple to update cache, but it is hard to deal with use case tests :-( So in 1-2 days I will finish it